.github/workflows/prepare.yaml

```diff
@@ -0,0 +1,49 @@
+name: Prepare environment
+
+on:
+  workflow_call:
+    null
+
+jobs:
+  build-or-load:
+    runs-on: ubuntu-latest
+
+    steps:
```

Comment:
Use the GitHub actions docker/setup-buildx-action and docker/build-push-action. Use GitHub Actions cache in build step without weekly restrictions.